clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureBaseClient.groovy

```diff
@@ -54,28 +54,71 @@ public abstract class AzureBaseClient {
     new ApplicationTokenCredentials(clientId, tenantId, secret, AzureEnvironment.AZURE)
   }
 
-  static Object getAzureOps(Closure getOps, String msgRetry, String msgFail, long count = AZURE_ATOMICOPERATION_RETRY) {
-    // The API call might return a timeout exception or some other Azure CloudException that is not the direct result of the operation
-    //   we are trying to execute; retry and if the final retry fails then throw
-    Object result = null
-    long operationRetry = 0
-    while (operationRetry < count) {
-      try {
-        operationRetry ++
-        result = getOps()
-        operationRetry = count
+  /**
+   * Wrap the call to an Azure operation to handle retry attempts
+   * @param operation - Operation to be execute
+   * @param count - number of retry attempts
+   * @return ServiceRespone returned from operation. If response results in 404 then return null
+   */
+  static <T> ServiceResponse<T> executeOp(Closure<ServiceResponse<T>> operation, long count = AZURE_ATOMICOPERATION_RETRY) {
+
+    long retryCount = count > 0 ? 0 : -1
```

Comment:
If you're not worried about the `count` arg being negative, can use: `long retryCount = count ? 0 : -1`